modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Pages/MongoPageRepository.cs

```diff
@@ -13,19 +18,48 @@ public MongoPageRepository(IMongoDbContextProvider<ICmsKitMongoDbContext> dbCont
         {
         }
 
-        public Task<Page> GetByUrlAsync(string url)
+        public virtual async Task<int> GetCountAsync(string filter = null, CancellationToken cancellationToken = default)
         {
-            return GetAsync(x => x.Url == url);
+            return await (await GetMongoQueryableAsync(GetCancellationToken(cancellationToken)))
```

Comment:
We can call `GetCancellationToken(cancellationToken)` once use returning value everywhere.